File: quarkus/manager-service/src/main/java/com/javanc/manager/domain/model/Company.java
package com.javanc.manager.domain.model;

import io.quarkus.mongodb.panache.common.MongoEntity;
import org.bson.codecs.pojo.annotations.BsonId;

import java.util.List;

@MongoEntity(collection = "company")
public class Company {
    @BsonId
    public Integer id;
    public String name;
    public String type;
    public String description;
    public String street;
    public String email;
    public String phone;
    public String city;
    public String country;
    public String url;
    public Integer idManager;
    public List<Integer> idHr;
    public List<Integer> idJobs;
}

File: client-react/src/types/company.ts
export interface CompanyDTO {
  id: number;
  name: string;
  type?: string;
  description?: string;
  street?: string;
  email?: string;
  phone?: string;
  city?: string;
  country?: string;
  idManager?: number;
  idHR?: number[];
  idJobs?: number[];
}

export interface CompanyFormValues {
  name: string;
  type?: string;
  description?: string;
  street?: string;
  email?: string;
  phone?: string;
  city?: string;
  country?: string;
  image?: File | null;
}
